k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeParameterEncoder.kt

```diff
@@ -0,0 +1,225 @@
+package com.fsck.k9.mail.internet
+
+import com.fsck.k9.mail.filter.Hex.appendHex
+import com.fsck.k9.mail.helper.encodeUtf8
+import com.fsck.k9.mail.helper.utf8Size
+
+/**
+ * Encode MIME parameter values as specified in RFC 2045 and RFC 2231.
```

Comment:
it's kind of ridiculous that we have to write code for this ourselves. fun fact though: googling for "[rfc2231 java library](https://www.google.de/search?q=rfc2231+java+library)" yields your github profile as the fifth hit (for me at least)